File: src/clients/python/examples/load_url.py
# /// script
# requires-python = ">=3.13"
# dependencies = [
#     "justbe-webview",
# ]
#
# [tool.uv.sources]
# justbe-webview = { path = "../" }
# ///
import asyncio

from justbe_webview import (
    WebView,
    Options,
    ContentUrl,
    Notification,
)


async def main():
    print("Creating webview")
    config = Options(
        title="Load Url Example",
        load=ContentUrl(
            url="https://example.com",
            headers={
                "Content-Type": "text/html",
            },
        ),
        userAgent="curl/7.81.0",
        devtools=True,
    )

    async with WebView(config) as webview:

        async def handle_start(event: Notification):
            await webview.open_devtools()
            await asyncio.sleep(2)  # Sleep for 2 seconds
            await webview.load_url(
                "https://val.town/",
                headers={
                    "Content-Type": "text/html",
                },
            )

        webview.on("started", handle_start)

    print("Webview closed")


if __name__ == "__main__":
    asyncio.run(main())

File: src/clients/go/examples/load_url.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/justbe-engineering/webview-client-go/webview"
)

func main() {
	ctx := context.Background()

	loadContent := webview.ContentFrom(webview.NewUrlContent("https://example.com", map[string]string{
		"Content-Type": "text/html",
	}))
	
	options := webview.Options{
		Title:     "Load Url Example",
		Devtools:  boolPtr(true),
		UserAgent: strPtr("curl/7.81.0"),
		Load: &loadContent,
	}

	wv, err := webview.NewWebView(ctx, options)
	if err != nil {
		log.Fatal(err)
	}
	defer wv.Close()

	// Register event handler for "started" event
	wv.On("started", func(event interface{}) {
		fmt.Println("WebView started!")

		// Open dev tools
		if err := wv.OpenDevTools(); err != nil {
			log.Printf("Failed to open dev tools: %v", err)
		}

		// Wait a bit then load a new URL
		time.Sleep(2 * time.Second)
		
		headers := map[string]string{
			"Content-Type": "text/html",
		}
		
		if err := wv.LoadURL("https://val.town/", headers); err != nil {
			log.Printf("Failed to load URL: %v", err)
		}
	})

	// Register event handler for window close
	wv.On("closed", func(event interface{}) {
		fmt.Println("WebView closed!")
	})

	// Wait for the webview to close
	if err := wv.Wait(); err != nil {
		log.Printf("WebView exited with error: %v", err)
	}
}

// Helper functions for pointer types
func boolPtr(b bool) *bool {
	return &b
}

func strPtr(s string) *string {
	return &s
}